src/main/scala/tilelink/ToAPB.scala

```diff
@@ -63,6 +63,10 @@ class TLToAPB(val aFlow: Boolean = true)(implicit p: Parameters) extends LazyMod
       val a_sel    = a.valid && RegNext(!in.d.valid || in.d.ready)
       val a_write  = edgeIn.hasData(a.bits)
 
+      val d_write  = RegNext(a_write)
+      val d_source = RegNext(a.bits.source)
+      val d_size   = RegNext(a.bits.size)
```

Comment:
Is the reason you aren't clock-gating these that they are tiny in practice?

Reply:
Also, the only criteria that would be safe is '!a_enable', which is true nearly all the time.

Reply:
Nevermind. I found a better condition.